### PR TITLE
Add lint for comparing to empty slices instead of using .is_empty()

### LINKED ISSUE #6217: Use is_empty instead of comparing to empty slice/str
### What it does

Suggest using `is_empty()` instead of checking for equality with an empty slice` == []` or `== ""` to be more consistent with similar lints such as `len_zero`

### Categories (optional)

- Kind: clippy::perf or clippy::style

*What is the advantage of the recommended code over the original code*

 - Consistency with similar lints such as `len_zero` so that we suggest lints for both `str.len() == 0` and `str == ""`
 - Performance wise the equality check will also usually take longer to compute compared to `is_empty()`

### Drawbacks

None.

### Example

```rust
if s == "" {
  ...
}

if arr == [] {
  ...
}
```

Could be written as:

```rust
if s.is_empty() {
  ...
}

if arr.is_empty() {
  ...
}
```

### PR DESCRIPTION
**Pull request by Urcra**
Hey first time making a clippy lint

I added the implementation of the lint the `len_zero` since it shared a lot of the code, I would otherwise have to rewrite. Just tell me if the lint should use it's own file instead

changelog: Add lint for comparing to empty slices

Fixes #6217

### CODE REVIEW

File: clippy_lints/src/len_zero.rs [hunk 1]
```diff
@@ -68,7 +68,44 @@ declare_clippy_lint! {
...
+    ///     ..
+    /// }
+    /// ```
+    /// Use instead:
+    /// ```ignore
+    /// if s.is_empty() {
+    ///     ..
+    /// }
+
+    /// if arr.is_empty() {
```
Comment (flip1995) [inline clippy_lints/src/len_zero.rs:99]:
```suggestion
    ///
    /// if arr == [] {
    ///     ..
    /// }
    /// ```
    /// Use instead:
    /// ```ignore
    /// if s.is_empty() {
    ///     ..
    /// }
    ///
    /// if arr.is_empty() {
```

**Review by flip1995 — APPROVED** [review]
Thanks for your (almost perfect) first contribution and welcome to Clippy! <3

**Comment by Urcra** [comment]
> Thanks for your (almost perfect) first contribution and welcome to Clippy! <3

Thanks, and thank you for the review

**Comment by flip1995** [comment]
@bors r+

Thanks!

**Pull request merged**